tox_core/src/dht/server_ext.rs

```diff
@@ -41,9 +40,8 @@ pub async fn dht_run_socket(
                 Err(e) => {
                     error!("packet receive error = {:?}", e);
                     // ignore packet decode errors
-                    if *e.kind() != DecodeErrorKind::Io { continue }
-                    else {
-                        return Err(Error::new(ErrorKind::Other, e.compat()))
+                    if let DecodeError::Io(e) = e {
+                        return Err(e);
```

Comment:
Continue?

Reply:
I inverted the condition - we ignore only decoding errors, not IO.